tests/ArrayDataMapTest.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+namespace Packaged\Map\Tests;
+
+use Packaged\Map\ArrayDataMap;
+use PHPUnit\Framework\TestCase;
+
+class ArrayDataMapTest extends TestCase
+{
+
+  public function testGet(): void
+  {
+    $array = new ArrayDataMap();
+
+    $array->set('a', '1');
+    $array->set('b', '2');
+    $array->set('c', 3);
+
+    self::assertEquals($array->get('a'), '1');
+    self::assertEquals($array->get('b'), '2');
+    self::assertEquals($array->get('c'), 3);
+
+    $array->set('d', ['x' => 'y', 'z']);
+    self::assertEquals($array->get('d'), ['a' => 'y', 'z']);
+  }
+
+  public function testAppend(): void
+  {
+    $data = ['a' => '1', 'b' => '2', 'c' => 3];
+    $arrayDataMap = new ArrayDataMap();
+
+    $arrayDataMap->set('1', $data);
+    $equals = [1 => $data];
```

Comment:
not a big fan of this equals array, esp when it gets to $equals[2].

Reply:
Can you provide an alternative? :)